### PR TITLE
kvutils - Remove unused protobuf field to prevent breaking changes 

### PR DESCRIPTION
**Pull request by nicu-da**
- the `submission_time` field was added as part of the 1.18 work but was later on replaced after the release branch was created
- to prevent backwards-incompatible changes and not store this field that is not used, easiest solution is to just remove it for the 1.18 release

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Review by SamirTalwar-DA — APPROVED** [review]
Seems sensible to me.

File: ledger/participant-state/kvutils/src/main/protobuf/com/daml/ledger/participant/state/kvutils/store/state.proto [hunk 1]
```diff
@@ -54,12 +54,10 @@ message DamlCommandDedupKey {
 
 message DamlCommandDedupValue {
   reserved 1; // was record_time
+  reserved 3; // was submission_time
```
Comment (gerolf-da) [inline ledger/participant-state/kvutils/src/main/protobuf/com/daml/ledger/participant/state/kvutils/store/state.proto:57]: If we introduced the `submission_time` field after SDK 1.17.x (that is, within the timeframe of 1.18), then there's no need to put it as "reserved" field, but you can just remove it.

Comment (gerolf-da) [reply ledger/participant-state/kvutils/src/main/protobuf/com/daml/ledger/participant/state/kvutils/store/state.proto:57]: Within a release, there's no need to preserve backwards compatibility.

Comment (nicu-da) [reply ledger/participant-state/kvutils/src/main/protobuf/com/daml/ledger/participant/state/kvutils/store/state.proto:57]: The field was introduced after 1.17.

If this is considered safe, and it's safe to break compatibility between 1.18 RC1 and 1.18 RC2 then there's actually no need for https://github.com/digital-asset/daml/pull/11781

**Pull request merged**